### ISSUE TITLE
Add an Isidedinventory for the Controller Block of Assembly Line and delete inventory from Regulator

### ISSUE THREAD

**Issue by ghost** [Blood-Asp/GT5-Unofficial#881]
If player want automate Assembly Line, he must have an opportunity to put and output for example data sticks in  controller block. Without this player haven't anyway to automate Assembly Line.

**Comment by leagris** [comment]
Automation for the assembly line is an impossible dream because the machine has not been designed for it.
You not only would need ability to automate recipes stick changes, but also have the controller filter what item stacks and amount goes into witch input bus at the right moment before starting production. And current Gregtech multi-block can't do this, as each one of buses and hatches works totally interdependently.
To be able to control input stacks and recipe stick changes, you would need to have all inputs and outputs directly at the controller block as it was done with Gregtech4 and earlier.

**Comment by codewarrior0** [comment]
I dunno. I think you can already automate the assembly line if your item/routing system is well-designed and can insert the items and fluids into the correct places. The only downside is needing a separate assembly line for each product and recipe stick.

Which isn't even a downside for some people!

**Comment by leagris** [comment]
You can certainly automate it for one fixed recipe, but if you change the recipe, you have to reconfigure your routing and filtering, so you already need player interaction to change the routing and filtering and it can't be changed remotely or without player interaction.

**Comment by ghost** [comment]
I'm create scheme of automation with company of enthusiasts and we understood that this is possible. Urge Blood Asp to fix this.

**Comment by OvermindDL1** [comment]
Could always add a button to the normal hatch's or make a new advanced version that has such a button to denote what 'slot' it represents in a recipe.  That could have quite useful uses elsewhere as well.

**Comment by leagris** [comment]
I know no automation system that can handle recipes patterns with more than 3x3 grid of single items or more than 2-3 ingredient stacks for processing in a machine.
The only mean I can think of; is using one of the computer mods like OpenComputers to select and drive right amount of ingredients to its respective input bus.

**Comment by codewarrior0** [comment]
Logistics Pipes can handle up to 9 processing ingredients plus any number of fluids.

**Comment by ghost** [comment]
I'm repeat that we have system which will be working with AE, but only if blood asp add Isidedinventory to controller.

**Comment by OvermindDL1** [comment]
> I'm repeat that we have system which will be working with AE, but only if blood asp add Isidedinventory to controller.

What would that accomplish though as (as far as I recall) the only inventory in the controller is the stack of machines that the Assembly Line works on?

**Comment by BurovB** [comment]
I didn't understand you

**Comment by codewarrior0** [comment]
> the stack of machines that the Assembly Line works on?

You're thinking of the Processing Array.

The Assembly Line's inventory has the data stick for the recipe it will use.

**Comment by OvermindDL1** [comment]
> You're thinking of the Processing Array.
> 
> The Assembly Line's inventory has the data stick for the recipe it will use.

Ah, indeed I was!  Then yes, why don't you submit a PR?  :-)

**Comment by codewarrior0** [comment]
Who, me? Because I don't actually mind building one more Assembly Line for each different recipe I want to use!

**Comment by ghost** [comment]
Provided that the ability to load, unload, add the controller line of how to work the system. The specifics of the line is such that if the bass input, say of size 3*3(because this is the limit of AE(though the ME interface provides only 8 slots, but this is enough)) throw things, then run the recipe on the line, the line uses just need at every stage of the ingredients, because of this, the method premises only need flash drives to the controller, it is possible to reduce the number of lines is 8 times(because of the capacity of ME interface). ME interfaces are connected to the controller, which pushes the right amount of things(just 9 types, as in the inventory of the regulator) to each input, but what would the regulator on the ate on the stack of things of each type that would need the controller turned off its own inventory. Things he will take from the entrance and his equipment will be the equipment to which it connected the entrance. Hence, if ME-interface thrust a card of creation and to establish the maintenance of the required number of items(for a single start line), then the inventory controller because of his lack fill will not, and the line will include items equal to 1 craft.

**Comment by leagris** [comment]
Assembly line is very close to real life. It is meant for batch processing. Configuring/reconfiguring a real-life assembly-line has significant cost.
In a factory, you configure your assembly lines, process batches of identical products and stock significant amount for future needs.
So the intended usage within Gregtech is: Build one assembly line, configure it (install recipe stick), probably automate supply of input buses. And run it for 1-3 stacks of the programmed product. You shall never need more than one 15 blocks long Assembly-Line, start it shorter if you are low in materials and up-size it as needed as your progress in Gregtech.

**Comment by ghost** [comment]
In this regard, I ask also Blood Asp to remove the inventory from the regulator.

**Comment by ghost** [comment]
> Assembly line is very close to real life. It is meant for batch processing. Configuring/reconfiguring a real-life assembly-line has significant cost.
> In a factory, you configure your assembly lines, process batches of identical products and stock significant amount for future needs.
> So the intended usage within Gregtech is: Build one assembly line, configure it (install recipe stick), probably automate supply of input buses. And run it for 1-3 stacks of the programmed product. You shall never need more than one 15 blocks long Assembly-Line, start it shorter if you are low in materials and up-size it as needed as your progress in Gregtech.

That's no reason not to automate Assembly Line

**Comment by ghost** [comment]
Another reason to remove the regulator inventory that it also can be used to automate a nuclear reactor without encountering the problem when the inventory controller are chock filled with cells.

**Comment by codewarrior0** [comment]
I don't understand what you mean about the regulator. But maybe you should open another issue for that?

The Assembly Line can already be automated. There's no need to make the recipe stick able to be inserted/extracted, unless you're cheap and only want to build one Assembly Line.

**Comment by ghost** [comment]
[http://ftb.gamepedia.com/Regulator](url)
It's regulator.

I don't want build fifteen or twenty assembly lines if i want automate them because it's stupid.

**Comment by dalekin** [comment]
Assembly line - multiblock, which by definition must be the same type of items to craft a mass character. Мass items - bars, rods, plates, chips, etc. We request the job and deliver materials.

If we add AE2-mod, the logic of an assembly line is organized with a plurality of AE2-interfaces, AE2-import buses, gregtech machines located at the desired location. Next, we can only get into the terminal and order the desired number of similar objects.

What do offer in GT assembly line to us? The machine is not able to distribute materials inside through one interface. The machine is not compatible with the AE2-network. The machine has recipes for items that do not belong to the massive detail - quantum armor. The machine can not be rearranged. Assembler line allows to pass through itself only one recipe.
To program 20 parts we need build 20 such lines - and we arrive at a situation where we start spamming gregtech machines, and not single but multiblock. We have evolved to the shooting range with the increase due to the more advanced machines to reduce lag, increase the crafting speed of items.

Seriously? We must spam multiblocks?

**Comment by leagris** [comment]
Stop this please!

What you want is not an assembly line, that's by definition is a set of machines disposed in serial operations, to efficiently perform batch assembly of a same item.
An assembly line with easy recipe change is not an assembly line. It has not been implemented for this purpose. And Yes! Building an assembly line per item type is a waste of resources, space and energy. How many LuV Motors, Robot Arms, Transmitters... would you ever need for your base?

If you think the assembly line is too costly, painful to automate...
Then, what you want is the older GT5.08 recipes system without the assembly line but with the Assembling Machine handling all component tiers.
And if you don't want an assembly line that require player to configure a recipe and produce large batches of same item; then bring back GT5.08 older recipes system by either installing Gregtech 5.08 and not 5.09, or use a tweaker script to restore the assembling machine recipes for those higher tier components.

**Comment by ghost** [comment]
If I create mod pack or blood asp will add more uses for Assembly Line we will have method for automate it. But it's stupid to don't realize automation only because it isn't widely used.
